src/transition.S

```diff
@@ -0,0 +1,62 @@
+.text
+.balign 4
+
+/* Wrapper to an intercept function, executed within a manager context;
+ * required in order to maintain a consisten execution state. This also manages
+ * setting the `DDC` as needed.
+ */
+.global intercept_wrapper
+.type intercept_wrapper, "function"
+// TODO restore c29
+intercept_wrapper:
+    mrs c28, ddc
+    msr ddc, c29
+    stp clr, c28, [sp, #-32]!
+    blr x10
```

Comment:
In C you declared this with one argument, which will be passed in `c0`. `x10` is not used for argument passing, which means calling this function from C will produce unpredictable results!

That's probably Ok, since we doing something _weird_ here anywhere (from C's perspective), but perhaps comment on the C declaration that it shouldn't be called, or something.

Reply:
Thanks for the observation. Added in https://github.com/capablevms/CHERI-ELF-comp/pull/2/commits/6265fbd2674993ec96173a13ee45cb2e7e1bd7d9.